### PR TITLE
Add generic Trial Plan

### PR DESCRIPTION
**Pull request by kjaksik**
**Description**

Changes proposed in this pull request:

- Add generic Trial Plan

**Related issue(s)**

### CODE REVIEW

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 1]
```diff
@@ -111,23 +92,30 @@ These are the provisioning parameters for GCP that you can configure:
...
-| **autoScalerMin**  | int    | Specifies the minimum number of virtual machines to create. | No | `3` |
-| **autoScalerMax**  | int    | Specifies the maximum number of virtual machines to create. | No | `3` |
-| **maxSurge**       | int    | Specifies the maximum number of virtual machines that are created during an update. | No | `1` |
-| **maxUnavailable** | int    | Specifies the maximum number of VMs that can be unavailable during an update. | No | `1` |
-     
-  </details>
-</div>
+| **name** | string | Specifies the provider-specific virtual machine type. | No | Any string| `n1-standard-4` |
+| **region** | int | Specifies the size of the root volume. | No | `europe`,`us` | `30` |
+| **provider** | string | Defines the cluster region. | No | `Azure`, `GCP` | `europe-west4` |
```
Comment (PK85) [inline docs/kyma-environment-broker/03-01-service-description.md:118]: adjust description, and I think you missed some columns, grid is not rendering correctly

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 2]
```diff
@@ -13,9 +13,8 @@ The supported plans are as follows:
 |-----------|-------------|
 | `azure` | Installs Kyma Runtime on the Azure cluster. |
 | `azure_lite` | Installs Kyma Lite on the Azure cluster. |
-| `azure_trial` | Installs Kyma Trial on the Azure cluster. |
 | `gcp` | Installs Kyma Runtime on the GCP cluster. |
-| `gcp_trial` | Installs Kyma Trial on the GCP cluster. |
+| `trial` | Installs Kyma Trial on chosen infrastructure. |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:17]:
```suggestion
| `trial` | Installs Kyma Trial on the chosen infrastructure. |
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 3]
```diff
@@ -111,23 +92,30 @@ These are the provisioning parameters for GCP that you can configure:
...
+
      
-  </details>
+## Trial Plan
+
+Trial Plan allows the user to choose the provider where they want to install Kyma. Trial features are that Kyma will be
+uninstalled, and the cluster will be deprovisioned after 30 days. Apart from that it's possible to provision only one Kyma
+per Global Account.
+
+These are the provisioning parameters for Trial Plan that you can configure:
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:106]:
```suggestion
These are the provisioning parameters for the Trial plan that you can configure:
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 4]
```diff
@@ -111,23 +92,30 @@ These are the provisioning parameters for GCP that you can configure:
...
+  
+<div tabs name="trial-plan" group="trial-plan">
   <details>
-  <summary label="gcp-trial-plan">
-  GCP Trial
+  <summary label="trial-plan">
+  Trial Plan
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:111]:
```suggestion
  Trial plan
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 5]
```diff
@@ -111,23 +92,30 @@ These are the provisioning parameters for GCP that you can configure:
 | **autoScalerMax** | int | Specifies the maximum number of virtual machines to create. | No | `4` |
 | **maxSurge** | int | Specifies the maximum number of virtual machines that are created during an update. | No | `4` |
 | **maxUnavailable** | int | Specifies the maximum number of VMs that can be unavailable during an update. | No | `1` |
+ 
+ </details>
+ </div>
+
      
-  </details>
+## Trial Plan
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:100]:
```suggestion
## Trial plan
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 6]
```diff
@@ -111,23 +92,30 @@ These are the provisioning parameters for GCP that you can configure:
...
+
+Trial Plan allows the user to choose the provider where they want to install Kyma. Trial features are that Kyma will be
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:102]:
```suggestion
Trial plan allows you to install Kyma on the infrastructure provided by any provider you choose. The Trial plan assumptions are as follows:
- Kyma is uninstalled after 30 days and the Kyma cluster is deprovisioned after this time.
- It's possible to provision only one Kyma Runtime per global account.
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 7]
```diff
@@ -111,23 +92,30 @@ These are the provisioning parameters for GCP that you can configure:
...
+uninstalled, and the cluster will be deprovisioned after 30 days. Apart from that it's possible to provision only one Kyma
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:103]:
```suggestion

```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 8]
```diff
@@ -111,23 +92,30 @@ These are the provisioning parameters for GCP that you can configure:
...
+per Global Account.
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:104]:
```suggestion

```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 9]
```diff
@@ -111,23 +92,30 @@ These are the provisioning parameters for GCP that you can configure:
...
-| **autoScalerMin**  | int    | Specifies the minimum number of virtual machines to create. | No | `3` |
-| **autoScalerMax**  | int    | Specifies the maximum number of virtual machines to create. | No | `3` |
-| **maxSurge**       | int    | Specifies the maximum number of virtual machines that are created during an update. | No | `1` |
-| **maxUnavailable** | int    | Specifies the maximum number of VMs that can be unavailable during an update. | No | `1` |
-     
-  </details>
-</div>
+| Parameter Name | Type | Description | Required | Possible values| Default value |  
+| ---------------|-------|-------------|----------|---------------|---------------|  
+| **name** | string | Specifies the name of the runtime. | Yes | Any string| None |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:116]:
```suggestion
| **name** | string | Specifies the name of the Kyma Runtime. | Yes | Any string| None |  
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 10]
```diff
@@ -111,23 +92,30 @@ These are the provisioning parameters for GCP that you can configure:
...
-| **maxSurge**       | int    | Specifies the maximum number of virtual machines that are created during an update. | No | `1` |
-| **maxUnavailable** | int    | Specifies the maximum number of VMs that can be unavailable during an update. | No | `1` |
-     
-  </details>
-</div>
+| Parameter Name | Type | Description | Required | Possible values| Default value |  
+| ---------------|-------|-------------|----------|---------------|---------------|  
+| **name** | string | Specifies the name of the runtime. | Yes | Any string| None |  
+| **region** | int | Defines the cluster region. | No | `europe`,`us` | `europe` |  
+| **provider** | string | Specifies the cloud provider used while provisioning. | No | `Azure`, `GCP` | `Azure` |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:118]:
```suggestion
| **provider** | string | Specifies the cloud provider used during provisioning. | No | `Azure`, `GCP` | `Azure` |
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 11]
```diff
@@ -111,23 +92,30 @@ These are the provisioning parameters for GCP that you can configure:
...
-| **region**         | string | Defines the cluster region. | No | `europe-west4` |
-| **zones**          | string | Defines the list of zones in which the Runtime Provisioner creates the cluster. | No | `["a", "b", "c"]` |
-| **autoScalerMin**  | int    | Specifies the minimum number of virtual machines to create. | No | `3` |
-| **autoScalerMax**  | int    | Specifies the maximum number of virtual machines to create. | No | `3` |
-| **maxSurge**       | int    | Specifies the maximum number of virtual machines that are created during an update. | No | `1` |
-| **maxUnavailable** | int    | Specifies the maximum number of VMs that can be unavailable during an update. | No | `1` |
-     
-  </details>
-</div>
+| Parameter Name | Type | Description | Required | Possible values| Default value |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:114]: Could you please correct this in all tables? 🥺 🙏

Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:114]:
```suggestion
| Parameter name | Type | Description | Required | Possible values| Default value |  
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 12]
```diff
@@ -13,9 +13,8 @@ The supported plans are as follows:
 |-----------|-------------|
 | `azure` | Installs Kyma Runtime on the Azure cluster. |
 | `azure_lite` | Installs Kyma Lite on the Azure cluster. |
-| `azure_trial` | Installs Kyma Trial on the Azure cluster. |
 | `gcp` | Installs Kyma Runtime on the GCP cluster. |
-| `gcp_trial` | Installs Kyma Trial on the GCP cluster. |
+| `trial` | Installs Kyma Trial on the chosen infrastructure. |
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:17]:
```suggestion
| `trial` | Installs Kyma Trial on Azure or GCP. |
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 13]
```diff
@@ -111,23 +92,30 @@ These are the provisioning parameters for GCP that you can configure:
...
 | **maxUnavailable** | int | Specifies the maximum number of VMs that can be unavailable during an update. | No | `1` |
+ 
+ </details>
+ </div>
+
      
-  </details>
+## Trial plan
+
+Trial plan allows you to install Kyma on the infrastructure provided by any provider you choose. The Trial plan assumptions are as follows:
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:102]:
```suggestion
Trial plan allows you to install Kyma either on Azure or GCP. The Trial plan assumptions are as follows:
```

**Review by klaudiagrz — APPROVED** [review]

File: components/kyma-environment-broker/internal/broker/plans.go [hunk 14]
```diff
@@ -229,75 +232,33 @@ func AzureSchema(machineTypes []string) []byte {
...
-                  "type": "string"
-                }
-              ]
-            }
-          },
-          "required": [
-            "name"
-          ]
-        }`
+  "$schema": "http://json-schema.org/draft-04/schema#",
```
Comment (jasiu001) [inline components/kyma-environment-broker/internal/broker/plans.go:237]: Why the schema isn't done similar to `AzureSchema` or `GCPSchema` as `RootSchema` struct?

File: docs/kyma-environment-broker/03-01-service-description.md (hunk 12 above)
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:17]:
```suggestion
| `trial` | Installs Kyma Trial on Azure or GCP. |
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 15]
```diff
@@ -111,23 +92,38 @@ These are the provisioning parameters for GCP that you can configure:
...
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:102]:
```suggestion
Trial plan allows you to install Kyma either on Azure or GCP. The Trial plan assumptions are as follows:
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 16]
```diff
@@ -111,23 +92,38 @@ These are the provisioning parameters for GCP that you can configure:
...
+- Kyma is uninstalled after 30 days and the Kyma cluster is deprovisioned after this time.
+- It's possible to provision only one Kyma Runtime per global account.
+
+To reduce the cost the Trial plan skips some of the [provisioning](./03-03-runtime-provisioning-and-deprovisioning.md#provisioning)
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:106]:
```suggestion
To reduce the cost, the Trial plan skips some of the [provisioning steps](./03-03-runtime-provisioning-and-deprovisioning.md#provisioning)
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 17]
```diff
@@ -111,23 +92,38 @@ These are the provisioning parameters for GCP that you can configure:
...
+steps.The skipped steps are:
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:107]:
```suggestion
These are the steps that are skipped during the Trial plan provisioning:
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 18]
```diff
@@ -111,23 +92,38 @@ These are the provisioning parameters for GCP that you can configure:
...
+- `Create_LMS_Tenant`
+- `Provision_Azure_Event_Hubs`
+- `Request_LMS_Certificates`
+
+Apart from these steps it also doesn't create the AVS External Evaluation
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:112]: Can't we include it in the list of steps?

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 19]
```diff
@@ -111,23 +92,37 @@ These are the provisioning parameters for GCP that you can configure:
...
+Trial plan allows you to install Kyma either on Azure or GCP. The Trial plan assumptions are as follows:
+- Kyma is uninstalled after 30 days and the Kyma cluster is deprovisioned after this time.
+- It's possible to provision only one Kyma Runtime per global account.
+
+To reduce the cost, the Trial plan skips some of the [provisioning steps](./03-03-runtime-provisioning-and-deprovisioning.md#provisioning)
+These are the steps that are skipped during the Trial plan provisioning:
+- `Create_LMS_Tenant`
+- `Provision_Azure_Event_Hubs`
+- `Request_LMS_Certificates`
+- `AVS External Evaluation` (part of the post actions during `Initialisation` step)
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:111]:
```suggestion
- `AVS External Evaluation` (part of the post actions during the `Initialisation` step)
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 20]
```diff
@@ -111,23 +92,37 @@ These are the provisioning parameters for GCP that you can configure:
...
+
      
-  </details>
+## Trial plan
+
+Trial plan allows you to install Kyma either on Azure or GCP. The Trial plan assumptions are as follows:
+- Kyma is uninstalled after 30 days and the Kyma cluster is deprovisioned after this time.
+- It's possible to provision only one Kyma Runtime per global account.
+
+To reduce the cost, the Trial plan skips some of the [provisioning steps](./03-03-runtime-provisioning-and-deprovisioning.md#provisioning)
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:106]:
```suggestion
To reduce the costs, the Trial plan skips some of the [provisioning steps](./03-03-runtime-provisioning-and-deprovisioning.md#provisioning)
```

File: docs/kyma-environment-broker/03-01-service-description.md [hunk 21]
```diff
@@ -111,23 +92,37 @@ These are the provisioning parameters for GCP that you can configure:
...
+These are the steps that are skipped during the Trial plan provisioning:
+- `Create_LMS_Tenant`
+- `Provision_Azure_Event_Hubs`
+- `Request_LMS_Certificates`
+- `AVS External Evaluation` (part of the post actions during `Initialisation` step)
+
+These are the provisioning parameters for the Trial plan that you can configure:
```
Comment (klaudiagrz) [inline docs/kyma-environment-broker/03-01-service-description.md:113]:
```suggestion
### Provisioning parameters

These are the provisioning parameters for the Trial plan that you can configure:
```

Comment (klaudiagrz) [reply docs/kyma-environment-broker/03-01-service-description.md:113]: What do you think of adding such a heading here? 🤔

**Review by jasiu001 — DISMISSED** [review]

**Review by jasiu001 — APPROVED** [review]

**Review by klaudiagrz — APPROVED** [review]
LGTM 🚀

**Review by PK85 — APPROVED** [review]

**Pull request merged**